google/cloud/ndb/_datastore_query.py

```diff
@@ -802,8 +816,38 @@ def _compare(self, other):
 
         return 0
 
+    def key(self):
+        """Construct the key for this result.
+
+        Returns:
+            key.Key: The key.
+        """
+        if self._key is None:
+            key_pb = self.result_pb.entity.key
+            ds_key = helpers.key_from_protobuf(key_pb)
+            self._key = key_module.Key._from_ds_key(ds_key)
+
+        return self._key
+
+    def check_cache(self, context):
+        """Check local context cache for entity.
+
+        Returns:
+            Any: The NDB entity for this result, if it is cached, otherwise
+                `_KEY_NOT_IN_CACHE`. May also return `None` if entity was deleted which
+                will cause `None` to be recorded in the cache.
```

Comment:
was the `None` return true of legacy as well then? Seems a bit odd but likely not problematic.

Reply:
Yes, this is [true of legacy](https://github.com/GoogleCloudPlatform/datastore-ndb-python/blob/master/ndb/context.py#L528) as well.